### PR TITLE
Swipe Quiz Analytics

### PR DESCRIPTION
**Pull request by jeremy6462**
This PR implements analytics for #18

### CODE REVIEW

File: AppBookDataAnalyzer/Analytics for Rows/Drawing/SketchPadDrawStrokeCount.swift [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    
+    func analyze(database: Database, textbookMaterial: TextbookMaterial) async -> String? {
+        
+        try? await database.pool.read { db in
+            
+            switch textbookMaterial {
+                
+            case .page(let appbook, let pageNumber):
+                
+                let query = Database.StrokeCollectionStroke.generateQuery(for: .draw, appbookId: appbook.id, pageNumber: pageNumber)
```
Comment (sdub18) [inline AppBookDataAnalyzer/Analytics for Rows/Drawing/SketchPadDrawStrokeCount.swift:23]: I feel like the syntax here is a little wonky. In your generate Query function the enum is called .draw, but hypothetically we could have multiple draw analytics (Draw Count, Draw Shape, Draw Stroke Weight, etc ...) Obviously we dont have those but I think just .draw or just .image can be a bit confusing as to the actual query itself.

**Review by sdub18 — APPROVED** [review]
Everything looks really good. One small syntax error worth looking at. But otherwise I'd say its great.

File: AppBookDataAnalyzer/Analytics for Rows/Swipe/Round One/SwipeableRoundOneLeft.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    
+    func analyze(database: Database, textbookMaterial: TextbookMaterial) async -> String? {
+        
+        try? await database.pool.read { db in
+            
+            switch textbookMaterial {
+                
+            case .page(let appbook, let pageNumber):
+                
+                let query = Database.SwipeableStudentAnswer.generateRoundOneQuery(isRightSwipe: false, appbookId: appbook.id, pageNumber: pageNumber)
```
Comment (sdub18) [inline AppBookDataAnalyzer/Analytics for Rows/Swipe/Round One/SwipeableRoundOneLeft.swift:23]: What is a round one query? Is that just a query for the first round of swipes? Like their initial reactions and not any redos they might do?

**Review by sdub18 — APPROVED** [review]
Looks great. The SQL statements seem really well explained. Im wondering if the return statement should be more than just a string? In python we can do return statements as a tuple of strings. While this adds a bit more layers to everything. It would be great to return say the `generateRoundOneQuery` in student answer with (response: String, questionId: String, and timestamp: String) as opposed to just string. Because then someone can really understand what they are getting from the function without having to open it up and dig through the SQL. Although tbh Im not sure if Swift supports this so may not be possible.

Anyway, food for thought!

File: AppBookDataAnalyzer/Analytics for Rows/Swipe/Round One/SwipeableRoundOneLeft.swift (hunk 2 above)
Comment (jeremy6462) [reply AppBookDataAnalyzer/Analytics for Rows/Swipe/Round One/SwipeableRoundOneLeft.swift:23]: Yea basically. In a swipe they first swipe on the options (round one query) then the can edit their responses (the changes query)

**Pull request merged**